crates/mun_hir/src/ty/tests.rs

```diff
@@ -1,10 +1,161 @@
 use crate::fixture::WithFixture;
 use crate::{
-    db::DefDatabase, diagnostics::DiagnosticSink, expr::BodySourceMap, mock::MockDatabase,
-    HirDisplay, InferenceResult, ModuleDef,
+    diagnostics::DiagnosticSink, expr::BodySourceMap, mock::MockDatabase, HirDisplay,
+    InferenceResult, ModuleDef, Package,
 };
 use std::{fmt::Write, sync::Arc};
 
+#[test]
+fn private_access() {
+    infer_snapshot(
+        r#"
+    //- /foo.mun
+    struct Foo {};
+    struct Bar(i64);
+    struct Baz;
+    fn foo() {}
+
+    pub struct PubFoo {};
+    pub struct PubBar(i64);
+    pub struct PubBaz;
+    pub fn pub_foo() {}
+
+    pub(super) struct PubSupFoo {};
+    pub(super) struct PubSupBar(i64);
+    pub(super) struct PubSupBaz;
+    pub(super) fn pub_sup_foo() {}
+
+    pub(package) struct PubPackageFoo {};
+    pub(package) struct PubPackageBar(i64);
+    pub(package) struct PubPackageBaz;
+    pub(package) fn pub_package_foo() {}
+
+    //- /bar.mun
```

Comment:
Does it make sense to add test cases for type aliases?

Reply:
I added them.